src/app/wallets/send-on-chain.ts

```diff
@@ -397,6 +382,51 @@ const executePaymentViaOnChain = async ({
     })
     if (journal instanceof Error) return journal
 
+    const txHash = await onChainService.payToAddress({
+      address,
+      amount: amountToSend,
+      targetConfirmations,
+      description: `journal-${journal.journalId}`,
+    })
+    if (
+      txHash instanceof InsufficientOnChainFundsError ||
+      txHash instanceof CPFPAncestorLimitReachedError
+    ) {
+      const reverted = await ledgerService.revertOnChainPayment({
+        journalId: journal.journalId,
+      })
+      if (reverted instanceof Error) return reverted
+      return txHash
+    }
+    if (txHash instanceof Error) {
+      logger.error(
+        { err: txHash, address, tokens: amountToSend, success: false },
+        "Impossible to sendToChainAddress",
+      )
+      return txHash
```

Comment:
Thinking about this again, I wonder if it makes sense to record a new critical exception here alongside the return. I'm thinking that if a critical alert gets thrown here it may not be immediately apparent that we should also check if the onchain payment actually failed and a manual reversion is needed.

Reply:
I'm just thinking also, maybe we can catch this in a new `updatePendingOnChainPayments` cronjob. We could:
- record the `txid` as a separate metadata field for successful transactions
- create a new `updatePendingOnChainPayments` method that cycles through pending onchain payments and uses their journalIds to look for onchain txns in our lnd that:
  - we can settle against if found
  - revert if not found

Reply:
`updatePendingOnChainPayments` is a pending task and will be added in another PR